benchmark/src/main/java/com/linecorp/decaton/benchmark/InProcessExecution.java

```diff
@@ -0,0 +1,144 @@
+
+import java.lang.management.CompilationMXBean;
+import java.lang.management.ManagementFactory;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import com.linecorp.decaton.benchmark.BenchmarkConfig.ProfilingConfig;
+import com.linecorp.decaton.benchmark.BenchmarkResult.JvmStats;
+import com.linecorp.decaton.benchmark.BenchmarkResult.Performance;
+import com.linecorp.decaton.benchmark.BenchmarkResult.ResourceUsage;
+import com.linecorp.decaton.benchmark.JvmTracker.GcStats;
+import com.linecorp.decaton.benchmark.ResourceTracker.TrackingValues;
+import com.linecorp.decaton.benchmark.Task.KafkaDeserializer;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * An {@link Execution} that executes benchmark within the instantiated JVM.
+ */
+@Slf4j
+public class InProcessExecution implements Execution {
+    private static final int COMPILE_TIME_CHECK_INTERVAL_MS = 5000;
+
+    private final boolean awaitJITStable;
+
+    public InProcessExecution(boolean awaitJITStable) {
+        this.awaitJITStable = awaitJITStable;
+    }
+
+    @Override
+    public BenchmarkResult execute(Config config, Consumer<Stage> stageCallback) throws InterruptedException {
+        BenchmarkConfig bmConfig = config.benchmarkConfig();
+
+        log.info("Loading runner {}", bmConfig.runner());
+        Runner runner = Runner.fromClassName(bmConfig.runner());
+        Recording recording = new Recording(bmConfig.tasks(), bmConfig.warmupTasks());
+        ResourceTracker resourceTracker = new ResourceTracker();
+        log.info("Initializing runner {}", bmConfig.runner());
+
+        Runner.Config runnerConfig = new Runner.Config(config.bootstrapServers(),
+                                                       config.topic(),
+                                                       new KafkaDeserializer(),
+                                                       bmConfig.params());
+        Profiling profiling = profiling(bmConfig);
+        runner.init(runnerConfig, recording, resourceTracker);
+        final Map<Long, TrackingValues> resourceUsageReport;
+        final Map<String, GcStats> jvmReport;
+        try {
+            stageCallback.accept(Stage.READY_WARMUP);
+            if (!recording.awaitWarmupComplete(3, TimeUnit.MINUTES)) {
+                throw new RuntimeException("timeout on awaiting benchmark to complete");
+            }
+            if (awaitJITStable) {
+                awaitJITGetsSettled();
+            }
+
+            profiling.start();
+            JvmTracker jvmTracker = JvmTracker.create();
+            stageCallback.accept(Stage.READY);
+            if (!recording.await(3, TimeUnit.MINUTES)) {
+                throw new RuntimeException("timeout on awaiting benchmark to complete");
+            }
+            profiling.stop().ifPresent(
+                    outputPath -> log.info("Profiling output is available at {}", outputPath));
```

Comment:
I think profiling output should be included in BenchmarkResult and be printed in [BenchmarkResult#print](https://github.com/line/decaton/pull/36/files#diff-a3e470608551e514a3285130e4b7b46dR135) because when we omit profiler output path, AsyncProfilerProfiling automatically determine output path and we have to find that from logs to see profile result.

Reply:
hm, that's a good point. 
The reason I didn't included it in the result is because I wanted to make `BenchmarkResult` portable, that is it might be serialized into JSON and brought around places like pasting the result making it not reasonable to contain contextual information like filepath.

but your point about the log is reasonble so, may I add a field to BenchmarkResult that doesn't gets serialized into JSON but yet printed in output?